release/tar/linux/opensearch-dashboards-tar-build.sh

```diff
@@ -0,0 +1,147 @@
+        :)
+            echo "Error: -${OPTARG} requires an argument"
+            usage
+            exit 1
+            ;;
+        ?)
+            echo "Invalid option: -${arg}"
+            exit 1
+            ;;
+    esac
+done
+
+if [ -z "$MANIFEST_FILE" ]; then
+    echo "error: You must specify the manifest file"
+    usage
+    exit 1
+fi
+
+if [ ! -f "$MANIFEST_FILE" ]; then
+    echo "error: The given file doesnot exist. Please check the specified path"
+    usage
+    exit 1
+fi
+
+SCHEMA_VERSION=`yq eval '.manifest.schema-version' $MANIFEST_FILE`
+PLATFORM=`yq eval '.build.platform' $MANIFEST_FILE`
+ARCHITECTURE=`yq eval '.build.architecture' $MANIFEST_FILE`
+VERSION=`yq eval '.build.version' $MANIFEST_FILE`
+UNDERLYING_ARCH=`uname -p`
+
+if [ -z "$VERSION" ]; then
+    echo "error: Please specify the version in the manifest file"
+    usage
+    exit 1
+fi
+
+if [ "$SCHEMA_VERSION" != "1.0" ]; then
+    echo "error: This script only supports manifest schema version 1.0. Please use the manifest file with right schema version"
+    usage
+    exit 1
+fi
+
+if [ "$PLATFORM" != "linux" ]; then
+    echo "error: $PLATFORM platform is not supported yet! Only supported platform is linux"
+    usage
+    exit 1
+fi
+
+if [ "$ARCHITECTURE" != "x64" ] && [ "$ARCHITECTURE" != "arm64" ]
+then
+    echo "error: $ARCHITECTURE architecture is not supported yet! Only supported architectures are x64 and arm64"
+    usage
+    exit 1
+fi
+
+if [[ ( "$ARCHITECTURE" == "arm64" && ( "$UNDERLYING_ARCH" != "aarch64" && "$UNDERLYING_ARCH" != "arm64" )) ||  ( "$ARCHITECTURE" == "x64" && ( "$UNDERLYING_ARCH" != "x86_64" && "$UNDERLYING_ARCH" != "amd64" )) ]]
+then
+    echo "error: The underlying architecture($UNDERLYING_ARCH) doesnot match the product architecture($ARCHITECTURE) that you are trying to build"
+    echo "Please use the appropriate build environment"
+    exit 1
+fi
+
+REPO_ROOT=`git rev-parse --show-toplevel`
+ROOT=`dirname $(realpath $0)`; echo $ROOT; cd $ROOT
+DIR_NAME="opensearch-dashboards-$VERSION"
+WORKING_DIR="$ROOT/$DIR_NAME"
+TARGET_DIR="$ROOT/target"
+PLUGINS_TEMP=`mktemp -d`
+OPENSEARCH_DASHBOARDS_CORE_URL=`yq eval '.products.opensearch-dashboards.opensearch-dashboards-min' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+OPENSEARCH_DASHBOARDS_PLUGINS_URLS=`yq eval '.products.opensearch-dashboards.plugins' $MANIFEST_FILE | sed s/^-// | sed -e 's/^[[:space:]]*//'`
+
+# # Download plugins to local 
+mkdir $WORKING_DIR
+mkdir -p $TARGET_DIR
+echo "working dir is: ${WORKING_DIR}"
+IFS=$'\n'
+for plugin in ${OPENSEARCH_DASHBOARDS_PLUGINS_URLS}; do
+  aws s3 cp $plugin $PLUGINS_TEMP/
+done
+
+# Download OpenSearch Dashboards min tarball
+wget -q ${OPENSEARCH_DASHBOARDS_CORE_URL} -O opensearch-dashboards.tar.gz
+tar -xzf opensearch-dashboards.tar.gz --strip-components=1 --directory "$WORKING_DIR" && rm -rf opensearch-dashboards.tar.gz
+
+# Install plugins to OpenSearch
+for plugins in $OPENSEARCH_DASHBOARDS_PLUGINS_URLS; do
```

Comment:
Very confusing use of variable in `plugins`, which is really just a single url of that plugin, use `plugin_url` is better.

Reply:
Changed